parmed/openmm/parameters.py

```diff
@@ -91,6 +91,32 @@ def __init__(self, *filenames):
         if filenames:
             raise NotImplementedError('Cannot yet read OpenMM Parameter sets')
 
+    @classmethod
+    def _remediate_residue_template(cls, params, residue):
+        """
+        Modify non-compliant residue templates to conform with OpenMM requirements.
+
+        * To correctly detect waters, OpenMM ffxml water models must not contain
+          non-chemical bond constraints. Theses are removed when importing
+          foreign parameter sets (e.g., CHARMM) into OpenMM parameter sets,
+          and not restored on conversion from OpenMM to other formats
+
+        """
+        # Populate atomic numbers in residue template
+        # TODO: This can be removed if the parameter readers are guaranteed to populate this correctly
+        for atom in residue.atoms:
+            if atom.type not in params.atom_types_str:
+                raise Exception('Residue {} contains atom type {} not found in parameter set'.format(residue.name, atom.type))
+            atom.atomic_number = params.atom_types_str[atom.type].atomic_number
+
+        # Check waters
+        if residue.empirical_chemical_formula == 'H2O':
+            # Remove any H-H bonds if they are present
+            for bond in residue.bonds:
+                if (bond.atom1.element_name == 'H') and (bond.atom2.element_name == 'H'):
+                    LOGGER.debug('Deleting H-H bond from water residue {}'.format(residue.name))
+                    residue.delete_bond(bond)
```

Comment:
You can't do this.  `residue.delete_bond` modifies `residue.bonds` as you're iterating over it.  Python doesn't like that.

You either need to loop over `list(residue.bonds)` or keep a list of bonds to remove from `self.bonds` after the iteration over all bonds is done.